storage/innobase/fil/fil0fil.cc

```diff
@@ -822,12 +822,19 @@ fil_node_open_file(
 			ut_error;
 		}
 
-		if (UNIV_UNLIKELY(space->flags != flags)) {
+		/* Validate the flags but do not compare the data directory
+		flag, in case this tablespace was relocated. */
+		const unsigned relevant_space_flags
+			= space->flags & ~FSP_FLAGS_MASK_DATA_DIR;
+		const unsigned relevant_flags
+			= flags & ~FSP_FLAGS_MASK_DATA_DIR;
+		if (UNIV_UNLIKELY(relevant_space_flags != relevant_flags)) {
```

Comment:
I would expect a warning if flag mis-match on data-dir part so that user know we incompatibility we are trying allowing.